completions/chflags

```diff
@@ -0,0 +1,31 @@
+# chflags(1) completion                                      -*- shell-script -*-
+
+_comp_cmd_chflags()
+{
+    local cur prev words cword was_split comp_args
+    _comp_initialize -- "$@" || return
+
+    if [[ $cur == -* ]]; then
+        # Complete -options
+        local w opts=""
+        for w in "${words[@]}"; do
+            [[ $w == -R ]] && opts="-H -L -P" && break
+        done
+        _comp_compgen -- -W '-f -h -v -R $opts'
+    else
+        local ret
+        # The first argument is a list of flags; the rest are filedir.
+        _comp_count_args
+        if ((ret == 1)); then
+            _comp_compgen -- -W 'simmutable nosimmutable sappend nosappend
+                                 archived noarchived opaque noopaque nodump
+                                 dump uimmutable nouimmutable uappend
+                                 nouappend hidden nohidden'
```

Comment:
The first argument seems to be a comma-separated list. Could we use `_comp_delimited` to support the completion of the comma-separated list? Also, does this list depend on the operating system? `chflags` of OpenBSD and NetBSD seem to define different sets of flags. Which implementations did you reference in preparing the list?

- [chflags(1) - OpenBSD manual pages](https://man.openbsd.org/chflags.1)
- [chflags(1)](https://man.freebsd.org/cgi/man.cgi?chflags(1))
- [chflags(1) - NetBSD Manual Pages](https://man.netbsd.org/NetBSD-9.0/chflags.1)

Reply:
> > Which implementations did you reference in preparing the list?
> 
> FreeBSD and Darwin.

Could you switch the list of flags based on the current system? E.g., using `[[ $OSTYPE == openbsd* ]]` and `[[ $OSTYPE == netbsd* ]]`.